### PR TITLE
feat(developers): added settings to block or forward email notifications

### PR DESCRIPTION
**Pull request by jeabakker**
Two plugin settings were added, one to be able to block email
notifications and one to forward all email notifications to a given
email address.

fixes: #3846, #10006

### CODE REVIEW

**Comment by jeabakker** [comment]
I also refactored the devtools form to use `elgg_view_field`

File: mod/developers/actions/developers/settings.php [hunk 1]
```diff
@@ -36,13 +36,13 @@
...
 foreach ($simple_settings as $setting) {
 	elgg_set_plugin_setting($setting, get_input($setting), 'developers');
 }
 
 elgg_flush_caches();
 
-system_message(elgg_echo('developers:settings:success'));
-
-forward(REFERER);
+return elgg_ok_response('', elgg_echo('developers:settings:success'), REFERER);
```
Comment (jdalsem) [inline mod/developers/actions/developers/settings.php]: no need for REFERER

File: mod/developers/classes/Elgg/DevelopersPlugin/Hooks.php [hunk 2]
```diff
@@ -78,4 +78,82 @@ public static function alterMenu($hook, $type, $output, $params) {
...
+		$email = $hook->getParam('email');
+		if (!($email instanceof \Elgg\Email)) {
+			return;
+		}
+		
+		$to = $email->getTo();
+		$users = get_user_by_email($to->getEmail());
+		if (empty($users)) {
+			// no user found, so this should be blocked
+			// as this e-mail address doesn't belong to an admin
```
Comment (jdalsem) [inline mod/developers/classes/Elgg/DevelopersPlugin/Hooks.php]: it doesn't belong to ANY user

File: mod/developers/languages/en.php [hunk 3]
```diff
@@ -33,8 +33,17 @@
...
 	'developers:label:show_gear' => "Use %s outside admin area",
 	'developers:help:show_gear' => "An icon on the bottom right of the viewport that gives admins access to developer settings and links.",
+	'developers:label:block_email' => "Block all outgoing e-mails",
+	'developers:help:block_email' => "You can block outgoing e-mail to regular users or to all users",
+	'developers:label:forward_email' => "Forward all outgoing e-mails to one address",
+	'developers:help:forward_email' => "All outgoing e-mails will be sent to the configured e-mail address",
+	
 	'developers:label:submit' => "Save and flush caches",
 
+	'developers:block_email:forward' => 'Forward all emails',
```
Comment (jdalsem) [inline mod/developers/languages/en.php]: inconsistent spelling of e-mail

File: mod/developers/start.php [hunk 4]
```diff
@@ -76,6 +76,16 @@ function developers_process_settings() {
 		$handler = [Hooks::class, 'alterMenu'];
 		elgg_register_plugin_hook_handler('view', 'navigation/menu/default', $handler);
 	}
+	
+	if (!empty($settings['block_email'])) {
```
Comment (jdalsem) [inline mod/developers/start.php]: why check? it is also checked in the hooks right?

File: mod/developers/views/default/admin/developers/settings.php [hunk 5]
```diff
@@ -5,82 +5,105 @@
 
 $data = [
```
Comment (jdalsem) [inline mod/developers/views/default/admin/developers/settings.php]: do these fields not belong in the form??

**Review by jdalsem — CHANGES_REQUESTED** [review]

File: mod/developers/start.php (hunk 4 above)
Comment (jeabakker) [reply mod/developers/start.php]: copied from the rest

File: mod/developers/views/default/admin/developers/settings.php (hunk 5 above)
Comment (jeabakker) [reply mod/developers/views/default/admin/developers/settings.php]: not going to refactor the entire plugin, just adding a setting ;)

**Comment by jdalsem** [comment]
also fixes #10006 ?

**Comment by jeabakker** [comment]
> also fixes #10006 ?

it offers a way to do that, yes. Set the forward address.

**Review by jdalsem — APPROVED** [review]

**Pull request merged**